src/Tribe/Views/V2/Views/By_Day_View.php

```diff
@@ -147,6 +148,73 @@ public function get_grid_days( $date = null, $force = false ) {
 				$repository = tribe_events( 'period' );
 			}
 			$repository->by_period( $grid_start_date, $grid_end_date )->fetch();
+		} else {
+			global $wpdb;
+
+			$first_grid_day = $days->start;
+			$start          = tribe_beginning_of_day( $first_grid_day->format( Dates::DBDATETIMEFORMAT ) );
+			$last_grid_day  = $days->end;
+			$end            = tribe_end_of_day( $last_grid_day->format( Dates::DBDATETIMEFORMAT ) );
+
+			$day_ids = tribe_events()
+				->set_found_rows( true )
+				->fields( 'ids' )
+				->by_args( $repository_args )
+				->where( 'date_overlaps', $start, $end, null, 2 )
+				->per_page( -1 )
+				->order_by( $order_by, $order )
+				->all();
+
+			$day_results = [];
+
+			$start_meta_key = '_EventStartDate';
+			$end_meta_key   = '_EventEndDate';
+
+			if ( Timezones::is_mode( 'site' ) ) {
+				$start_meta_key = '_EventStartDateUTC';
+				$end_meta_key   = '_EventEndDateUTC';
+			}
+
+			$results = [];
+			$request_chunks = array_chunk( $day_ids, 200 );
+
+			foreach ( $request_chunks as $chunk_ids ) {
+				$sql = "
+				SELECT
+				  post_id,
+					meta_key,
+					meta_value
+				FROM
+					{$wpdb->postmeta}
+				WHERE
+					meta_key IN ( %s, %s )
+					AND post_id IN ( " . implode( ',', $chunk_ids ) . " )
+			";
+
+				$chunk_results = $wpdb->get_results( $wpdb->prepare( $sql, [ $start_meta_key, $end_meta_key ] ) );
```

Comment:
* :warning: Usage of a direct database call is discouraged.
* :warning: Direct database call without caching detected. Consider using wp_cache_get() / wp_cache_set() or wp_cache_delete().
* :no_entry_sign: Use placeholders and $wpdb->prepare(); found $sql
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: c13180fc078549e699fadc24a06622ea</sub>